Sources/Utility/ArgumentParser.swift

```diff
@@ -509,27 +639,30 @@ public final class ArgumentParser {
     private func parse(_ args: [String] = [], parent: Result?) throws -> Result {
         let result = Result(parser: self, parent: parent)
         // Create options map to quickly look up the arguments.
-        let optionsTuple = options.flatMap({ option -> [(String, AnyArgument)] in
-            var result = [(option.name, option)]
-            // Add the short names too, if we have them.
-            if let shortName = option.shortName {
-                result += [(shortName, option)]
+        // FIXME: Replace with inout reduce once available
+        var optionsMap: [String:ParserOptionArgument] = [:]
+        for option in optionArguments {
+            optionsMap[option.argument.name] = option
+            if let shortName = option.argument.shortName {
+                optionsMap[shortName] = option
             }
-            return result
-        })
-        let optionsMap = Dictionary(items: optionsTuple)
+        }
+
         // Create iterators.
-        var positionalArgsIterator = positionalArgs.makeIterator()
+        var positionalIterator = positionalArguments.makeIterator()
         var argumentsIterator = args.makeIterator()
 
         while let arg = argumentsIterator.next() {
+            let parserArgument: ParserArgument
+            var parser: Parser
+
             // If argument is help then just print usage and exit.
             if arg == "-h" || arg == "-help" || arg == "--help" {
                 printUsage(on: stdoutStream)
                 exit(0)
             } else if isPositional(argument: arg) {
                 /// If this parser has subparsers, we allow only one positional argument which is the subparser command.
-                if !subparsers.isEmpty {
+                guard subparsers.isEmpty else {
```

Comment:
Yes, I agree guard is good for early returns but I tend to avoid it when there are more than two statements in body of guard. I don't really mind the change but please commit it as NFC.